src/yb/master/catalog_manager_bg_tasks.cc

```diff
@@ -175,6 +175,9 @@ void CatalogManagerBgTasks::Run() {
 
       // Start the tablespace background task.
       catalog_manager_->StartTablespaceBgTaskIfStopped();
+    } else {
```

Comment:
I think there should be a `if (!l.catalog_status().ok())` clause above this, I think its best to put this there.

Reply:
when (!l.catalog_status().ok()) is true,  catalog-manager status is either kConstructed or kStarting, it means catalog manager is init that metrics already reset, we should do nothing. we only need to clear metrics when raft role change from leader to other state so that reset metrics in branch "else"

Reply:
Sorry for the late reply, that's a fair point. The code looks good and I will put it through our jenkins pipeline.

Reply:
thanks